src/Adapter/Product/Repository/TagRepository.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Repository;
+
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotUpdateProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\LocalizedTags;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShopException;
+use Tag;
+
+/**
+ * Accesses product Tag data source
+ */
+class TagRepository
+{
+    public function addTags(ProductId $productId, LanguageId $languageId, LocalizedTags $localizedTags): void
```

Comment:
```suggestion
    public function addTagsByLanguage(ProductId $productId, LanguageId $languageId, LocalizedTags $localizedTags): void
```

Or `addTagsInLang` since it only handles one Language at a time the function should reflect this behaviour, because at first I expected it to handle all languages in a loop. Also why not using only one parameter `LocalizedTags` since it also contains the `languageId` so it increases the number of parameter for no reason.

Unless we want to avoid using `LocalizedTags` in the repository to keep simpler input parameters which does make sense, but in this case the fourth parameter could be an array, the the updater is to only one to handle this `LocalizedTags` object and would call: `$this->tagRepository->addTags($productId, $localizedTags->getLanguageId(), $localizedTags-> getTags());`